Tests/Command/CleanMediaCommandTest.php

```diff
@@ -276,8 +288,39 @@ public function testExecute()
 
         $output = $this->tester->execute(array('command' => $this->command->getName()));
 
-        $this->assertRegExp('@Context: foo\s+\'qwertz.ext\' was successfully removed\s+\'thumb_1_bar.ext\' was successfully removed\s+done!@', $this->tester->getDisplay());
-
+        $this->assertOutputFoundInContext(
+            '/Context: foo\s+(.+)\s+done!/ms',
+            array(
+                '\'qwertz.ext\' was successfully removed',
+                '\'thumb_1_bar.ext\' was successfully removed',
+            ),
+            $this->tester->getDisplay()
+        );
         $this->assertSame(0, $output);
     }
+
+    /**
+     * Asserts whether all expected texts can be found in the output within a given context.
+     *
+     * @param string $extract  PCRE regex expected to have a single matching group, extracting the content of a context
+     * @param array  $expected Excerpts of text expected to be found in the output
+     * @param string $output   Output to find
+     */
+    private function assertOutputFoundInContext($extractor, $expected, $output)
+    {
+        preg_match_all($extractor, $output, $matches);
+
+        $found = false;
+        foreach ($matches[1] as $match) {
+            foreach ($expected as $excerpt) {
+                if (false === strpos($match, $excerpt)) {
+                    continue 2;
```

Comment:
It's breaking the innermost `foreach` (`foreach ($expected`) and continue the enclosing one (`foreach ($matches[1]`).

If you feel it's a hard-to-read or esoteric syntax, I could probably replace it by something better, although I'm not sure what (additional private method maybe)...